### PR TITLE
add spinner for when images are being captured

### LINKED ISSUE #78: Add spinner for when images are being captured
## Context

We want to provide the user with a visual indication that their screenshot is in progress

### PR DESCRIPTION
**Pull request by will-weiss**
Closes #78 . We use a number to represent `addingImages` in case someone hits the button several times. It looks decent when there's already another image present but bad for the first image. I say we get this in and delay that until #80

image
image

### CODE REVIEW

File: scss/popup.scss [hunk 1]
```diff
@@ -1,9 +1,10 @@
 @use '../node_modules/draft-js/dist/Draft.css';
 @use '../node_modules/emoji-mart/css/emoji-mart.css';
-@use './components/images.scss';
+@import '../node_modules/spinthatshit/src/loaders.scss';
+@import './colors';
+@import './components/images.scss';
```
Comment (shachar-langer) [inline scss/popup.scss:5]: Why not using `use`?

File: src/background/responders.ts [hunk 2]
```diff
@@ -73,7 +87,7 @@ export const responders: Responders<Action> = {
   updateEditorState(state, { editorState }): Partial<StoreState> {
     return updateActiveFeedback(state, () => ({ editorState }))
   },
-  clickTakeScreenshot(): Partial<StoreState> {
+  clickTakeScreenshot(state): Partial<StoreState> {
```
Comment (shachar-langer) [inline src/background/responders.ts:90]: Redundant

File: src/background/responders.ts [hunk 3]
```diff
@@ -102,16 +116,25 @@ export const responders: Responders<Action> = {
       }),
     }
   },
+  imageCaptureStart(state, { targetId }): Partial<StoreState> {
+    return updateFeedbackByTargetId(state, targetId, target => ({
+      addingImages: target.feedbackState.addingImages + 1,
+    }))
+  },
```
Comment (shachar-langer) [inline src/background/responders.ts:123]: Clever.

File: src/popup/components/images.tsx [hunk 4]
```diff
@@ -48,11 +48,22 @@ export function ImageThumbnail({ image, startEditingImage, clickDeleteImage, del
...
+  )
+}
+
 export function Images({ feedback, startEditingImage, clickDeleteImage, deleteImageDisabled }: ImagesProps): null | JSX.Element {
-  if (!feedback.images.length) return null
+  if (!feedback.images.length && !feedback.addingImages) return null
 
   return (
     <div className="images">
+      {!!feedback.addingImages && <ImageSpinner />}
```
Comment (shachar-langer) [inline src/popup/components/images.tsx:66]: If `addingImages` is bigger than 1, we should render more `ImageSpinner`s.

File: src/popup/components/images.tsx [hunk 5]
```diff
@@ -48,11 +48,22 @@ export function ImageThumbnail({ image, startEditingImage, clickDeleteImage, del
...
       {feedback.images.map((image, index) => (
```
Comment (shachar-langer) [inline src/popup/components/images.tsx:67]: How do we know not to post a tweet until all images were added? How do we handle the situation when a user wants to abort the process? I think we should still have an 'X' option

File: src/tests/mocks.tsx [hunk 6]
```diff
@@ -31,10 +33,18 @@ export function createMocks(): Mocks {
   const popupWindow = new JSDOM(popupHTML).window
   popupWindow.roarServerUrl = 'https://test-roar-server.com'
 
+  const captureVisibleTabResolvers: any[] = []
+  const resolveLatestCaptureVisibleTab = () => {
```
Comment (shachar-langer) [inline src/tests/mocks.tsx:37]: 👍

**Comment by shachar-langer** [comment]
This PR also resolves [this](https://github.com/morehumaninternet/roar-extension/issues/39) issue

File: scss/popup.scss (hunk 1 above)
Comment (will-weiss) [reply scss/popup.scss:5]: use is for external css that you want to inline, import allows you to refer to scss variables

File: src/popup/components/images.tsx (hunk 5 above)
Comment (will-weiss) [reply src/popup/components/images.tsx:67]: Great call, adding that

**Review by shachar-langer — APPROVED** [review]

**Pull request merged**